disperser/controller/dispatcher.go

```diff
@@ -381,64 +410,90 @@ func (d *Dispatcher) HandleSignatures(
-			return fmt.Errorf("failed to update blob statuses for batch %s: %w", batchHeaderHash, err)
-		}
-		return fmt.Errorf("all quorums received no attestation for batch %s", batchHeaderHash)
+	return nil
+}
+
+// submitAttestation submits a QuorumAttestation to the blobMetadataStore
+func (d *Dispatcher) submitAttestation(
+	ctx context.Context,
+	batchData *batchData,
+	quorumAttestation *core.QuorumAttestation,
+) error {
+	sortedNonZeroQuorums, quorumPercentages := d.parseAndLogQuorumPercentages(
+		hex.EncodeToString(batchData.BatchHeaderHash[:]),
+		quorumAttestation.QuorumResults)
+	if len(sortedNonZeroQuorums) == 0 {
+		return errors.New("all quorums received no attestation for batch")
 	}
-	slices.Sort(nonZeroQuorums)
 
-	aggSig, err := d.aggregator.AggregateSignatures(ctx, d.chainState,
-		uint(batchData.Batch.BatchHeader.ReferenceBlockNumber), quorumAttestation, nonZeroQuorums)
-	aggregateSignaturesFinished := time.Now()
-	d.metrics.reportAggregateSignaturesLatency(aggregateSignaturesFinished.Sub(receiveSignaturesFinished))
+	aggregationStartTime := time.Now()
+	signatureAggregation, err := d.aggregator.AggregateSignatures(
+		ctx,
+		d.chainState,
+		uint(batchData.Batch.BatchHeader.ReferenceBlockNumber),
+		quorumAttestation,
+		sortedNonZeroQuorums)
+	d.metrics.reportAggregateSignaturesLatency(time.Since(aggregationStartTime))
 	if err != nil {
-		dbErr := d.failBatch(ctx, batchData)
-		if dbErr != nil {
-			return fmt.Errorf("failed to update blob statuses for batch %s to failed: %w", batchHeaderHash, dbErr)
-		}
-		return fmt.Errorf("failed to aggregate signatures for batch %s: %w", batchHeaderHash, err)
+		return fmt.Errorf("aggregate signatures: %w", err)
 	}
 
-	attestation = &corev2.Attestation{
+	attestation := &corev2.Attestation{
 		BatchHeader:      batchData.Batch.BatchHeader,
 		AttestedAt:       uint64(time.Now().UnixNano()),
-		NonSignerPubKeys: aggSig.NonSigners,
-		APKG2:            aggSig.AggPubKey,
-		QuorumAPKs:       aggSig.QuorumAggPubKeys,
-		Sigma:            aggSig.AggSignature,
-		QuorumNumbers:    nonZeroQuorums,
-		QuorumResults:    quorumResults,
+		NonSignerPubKeys: signatureAggregation.NonSigners,
+		APKG2:            signatureAggregation.AggPubKey,
+		QuorumAPKs:       signatureAggregation.QuorumAggPubKeys,
+		Sigma:            signatureAggregation.AggSignature,
+		QuorumNumbers:    sortedNonZeroQuorums,
+		QuorumResults:    quorumPercentages,
 	}
+
+	putAttestationStartTime := time.Now()
 	err = d.blobMetadataStore.PutAttestation(ctx, attestation)
-	putAttestationFinished := time.Now()
-	d.metrics.reportPutAttestationLatency(putAttestationFinished.Sub(aggregateSignaturesFinished))
 	if err != nil {
-		dbErr := d.failBatch(ctx, batchData)
-		if dbErr != nil {
-			return fmt.Errorf("failed to update blob statuses for batch %s to failed: %w", batchHeaderHash, dbErr)
-		}
-		return fmt.Errorf("failed to put attestation for batch %s: %w", batchHeaderHash, err)
+		return fmt.Errorf("put attestation: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("error calling put attestation: %w", err)
```

Reply:
done

Reply:
No need to revert as this is just a nit but overall @cody-littley I prefer not to encode redundant information in errors that are getting wrapped. This follows the uber convention: https://github.com/uber-go/guide/blob/master/style.md#error-wrapping

I'm totally onboard with adding the "error" prefix in logs though! Because those are not wrapped and are printed as is, so extra context is useful

Reply:
I agree with not prefixing errors this way- I misread the line and thought this was a log when applying this change.

Reverted for consistency

Reply:
In that case, I'm ok if you don't prefix the error string (here as well as elsewhere in this PR)